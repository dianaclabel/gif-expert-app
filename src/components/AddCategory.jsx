import React, { useState } from "react";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("One punch");

  const onInputChange = ({ target }) => {
    console.log(target.value);
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;

    setCategories((categories) => [...categories, inputValue]);
    setInputValue(" ");
  };

  return (
    <form onSubmit={(event) => onSubmit(event)}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
