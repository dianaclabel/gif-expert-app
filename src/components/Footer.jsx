export const Footer = () => {
  return (
    <div className="footer">
      <span>Creado by Diana Clabel 2024</span>
    </div>
  );
};
