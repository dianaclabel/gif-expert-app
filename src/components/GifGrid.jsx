import { GifItem } from "./Gifitem.jsx";
import { useFetchGifs } from "../hooks/useFetchGifs.js";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  console.log({ isLoading });

  return (
    <>
      <h3>{category}</h3>

      {isLoading && <h2>cargando...</h2>}

      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>

      {/* <button onClick={() => setCounter(counter + 1)}> +1</button> */}
    </>
  );
};
