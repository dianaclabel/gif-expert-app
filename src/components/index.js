export * from "./AddCategory";
export * from "./Gifitem";
export * from "./GifGrid";
