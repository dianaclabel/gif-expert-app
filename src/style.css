* {
  font-family: Helvetica, Arial, sans-serif;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  background-color: #ead196;
}

.title-principal {
  color: #bf3131;
  font-size: 3rem;
  padding-bottom: 2rem;
  text-align: center;
}

form {
  display: flex;
  justify-content: center;
}

input {
  background-color: white;
  border-radius: 10px;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 80%;
  border: none;
}

h2 {
  font-size: 1.5rem;
}

.sub-title {
  font-size: 2.5rem;
  padding: 15px;
  color: #a0153e;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
  background-color: rgb(24, 18, 18);
}

.card p {
  background-color: rgb(24, 18, 18);
  color: rgb(219, 226, 231);
  flex: 1;
  font-size: 1.5rem;
  margin-top: 5px;
  /* padding: 5px 20px 0px 20px; */
  padding: 20px;
  text-align: center;
}

.card img {
  width: 100%;
}

.footer {
  background-color: rgb(24, 18, 18);
  color: rgb(252, 254, 255);
  width: 100%;
  text-align: center;
  padding: 20px;
}
