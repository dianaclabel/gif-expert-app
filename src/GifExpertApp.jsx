import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

// import React from "react";
// Despues de la version 17 de react ya no es necesario estar importando react a cada uno.

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch", "Dragon Ball"]);

  const onAddCategory = () => {
    setCategories([...categories, "Valorant"]);
  };

  console.log(categories);
  return (
    <div>
      <>
        {/* titulo */}
        <h1>GifExpertApp</h1>
        {/* input */}
        <AddCategory />
        {/* listado de gif */}
        <button onClick={onAddCategory}>Agregar</button>
        <ol>
          {categories.map((category) => {
            return <li key={category}>{category}</li>;
          })}
          {/* <li>ABC</li> */}
        </ol>
        {/* gif item */}
      </>
    </div>
  );
};
