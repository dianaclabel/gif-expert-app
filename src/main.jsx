import React from "react";
import ReactDOM from "react-dom/client";
import { GifExpertApp } from "./GifExpertApp";
import { Footer } from "./components/Footer";
import "./style.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  // <React.StrictMode>
  <>
    <GifExpertApp />
    <Footer></Footer>
  </>
  // </React.StrictMode>
);
