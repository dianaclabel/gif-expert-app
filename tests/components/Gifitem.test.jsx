import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("Pruebas de componentes", () => {
  const title = "One punch";
  const url =
    "https://media4.giphy.com/media/Qm3xCXWifrgzb9aEqg/giphy.gif?cid=42367ee0x2m6sj8mfrvlwpus1n00rc6n1srgf79b6b01r7hv&ep=v1_gifs_search&rid=giphy.gif&ct=g";

  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el  URL y el ALT indicado ", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("Debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
