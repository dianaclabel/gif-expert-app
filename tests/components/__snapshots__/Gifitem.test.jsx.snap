// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componentes Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One punch"
      src="https://media4.giphy.com/media/Qm3xCXWifrgzb9aEqg/giphy.gif?cid=42367ee0x2m6sj8mfrvlwpus1n00rc6n1srgf79b6b01r7hv&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      One punch
    </p>
  </div>
</div>
`;
